vizro-core/src/vizro/actions/_collapse_expand_container.py

```diff
@@ -0,0 +1,58 @@
+from typing import Any, Literal
+
+from pydantic import Field, model_validator
+
+from vizro.actions._abstract_action import _AbstractAction
+from vizro.managers import model_manager
+from vizro.models._models_utils import _log_call
+from vizro.models.types import ModelID
+
+
+class collapse_expand_containers(_AbstractAction):
+    """Collapses and/or expands selected collapsible containers.
+
+    Args:
+        collapse (list[ModelID]): List of collapsible container ids to collapse. Defaults to `[]`.
+        expand (list[ModelID]): List of collapsible container ids to expand. Defaults to `[]`.
+    """
+
+    type: Literal["collapse_expand_containers"] = "collapse_expand_containers"
+    collapse: list[ModelID] = Field(default=[], description="List of collapsible container ids to collapse.")
+    expand: list[ModelID] = Field(default=[], description="List of collapsible container ids to expand.")
+    # TODO-AV2 D 5: toggle would need multiple state as input. Make that built-in actions can deal with arbitrary
+    #  number of inputs.
+    # toggle: list[ModelID]
+
+    @model_validator(mode="after")
+    def validate_collapse_and_expand(self):
+        if not self.collapse and not self.expand:
+            raise ValueError("Either the `collapse` or `expand` list must contain at least a single element.")
+        if set(self.collapse) & set(self.expand):
+            raise ValueError("Collapse and expand lists cannot contain the same elements!")
```

Comment:
```suggestion
        if (overlap := set(self.collapse) & set(self.expand)):
            raise ValueError(f"`collapse` and `expand` cannot both contain the same IDs {overlap}.")
```

Just to be a bit more explicit with the error message and identify the duplicated IDs.